### PR TITLE
QTest: Add ability to preserve user-specified environment variables

### PR DESCRIPTION
**Pull request by anirudhsanthiar**

### CODE REVIEW

**Comment by narasamdya** [comment]
Please provide a more descriptive title and a better description (as well as associate the change with an assigned work item in azdev). Or, you can simply make this PR a draft one.

File: Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc [hunk 1]
```diff
@@ -170,6 +170,8 @@ export function runQTest(args: QTestArguments): Result {
         Cmd.flag("--enableVsJitDebugger", Environment.hasVariable("[Sdk.BuildXL]enableVsJitDebugger")),
         Cmd.flag("--qTestIgnoreQTestSkip", args.qTestIgnoreQTestSkip),
         Cmd.option("--qTestAdditionalOptions ", args.qTestAdditionalOptions, args.qTestAdditionalOptions ? true : false),
+        Cmd.option("--envVarsToKeep ", args.envVarsToKeep || "")
+    ];
```
Comment (narasamdya) [inline Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc]: > ]; [](start = 3, length = 3)

I'm not sure if this is a valid syntax.

File: Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc [hunk 2]
```diff
@@ -170,6 +170,7 @@ export function runQTest(args: QTestArguments): Result {
         Cmd.flag("--enableVsJitDebugger", Environment.hasVariable("[Sdk.BuildXL]enableVsJitDebugger")),
         Cmd.flag("--qTestIgnoreQTestSkip", args.qTestIgnoreQTestSkip),
         Cmd.option("--qTestAdditionalOptions ", args.qTestAdditionalOptions, args.qTestAdditionalOptions ? true : false),
+        Cmd.option("--envVarsToKeep ", args.envVarsToKeep || ""),
```
Comment (surthumm) [inline Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc]: >|| "" [](start = 58, length = 5)

You dont need this part. If args.envVarsToKeep is null, BuildXL ignores adding this argument. You can validate it though.

Comment (mpysson) [inline Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc]: BuildXL already limits the environment variables visible to its child processes. Is an additional QTest restriction necessary? I worry about adding configuration complexity here. What would be the harm of allowing all of the DBS.QTest.exe pip's environment variables to pass through? Maybe you can provide an example to illustrate the use case

**Comment by mpysson** [comment]
>         environmentVariables: [{ name: "[Sdk.BuildXL]qCodeCoverageEnumType", value: qCodeCoverageEnumType }],

If these are user specified environment variables, you'll need to specify them here too for BuildXL to actually pass them to DBS.QTest.exe 

---
Refers to: Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc:198 in 633094d. [](commit_id = 633094df6ef105e57443a62f20b8fd3ba2044f91, deletion_comment = False)

File: Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc (hunk 2 above)
Comment (anirudhsanthiar) [reply Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc]: Suresh has a PR that will allow all of the pip environment variables through. Let me abandon this PR for now.

**Pull request closed without merging**